### DISCUSSION TITLE
Get one document via POST route

### DISCUSSION THREAD

**Discussion by ghost** [meilisearch/product#464] [Feedback & Feature Proposal]
Currently you can only 'get one document' via the GET route.

If using api authentication, fetching a document using GET requires a CORS pre-flight request.  This degrades performance as this pre-flight can not be cached.

You should be able to 'get one document', or' get documents', via the POST route so the pre-flight request can be cached for improved performance.  You can search via the POST route for this very reason, and I think it should be extended for fetching documents.

**Comment by gmourier** [comment]
Hi @jordanchubb 👋

Can you please tell us more about your use case and why you need to fetch one document your case?

Thanks!

**Reply by ghost** [reply]
The use case is a public API with an instantsearch frontend.
The main reason I would like this is because the documents in my index can be quite large.
If I make a search (getting all attributes) it is up to 500kb per request. If I make the same search with limited attributes (via attributesToRetrieve) it is around 5kb per request. Because of instantsearch, each character typed into the search bar produces a request.  By only getting title and description attributes, the frontend is significantly quicker.  From there, if a user interacts with a search result, I then make a request for the relevant document with all attributes.

**Reply by gmourier** [reply]
Hi @jordanchubb 👋

Just to be sure about your workflow (Correct me if I'm wrong, I'm trying some extrapolation here)

- A user makes a search (where the `attributesToRetrieve` are minimal to reduce the network size)
- Then you want to use `GET /indexes/yourIndex/documents/theSelectedDocument` to load the specific document page with all the document attributes?

For the last step; instead of querying it from `GET /indexes/yourIndex/documents/theSelectedDocument` why don't you use a POST search request in addition to a filter on the document identifier? e.g `"filter": "id = :selectedId"`

Thanks!
